docs/contributors/conventions.rst

```diff
@@ -0,0 +1,118 @@
+.. _coding conventions:
+
+##################
+Coding conventions
+##################
+
+Python
+------
+
+.. dropdown:: TL;DR
+
+    We use ``black`` to format our code, ``flake8`` to lint it, and ``isort`` to sort our imports.
+
+    .. important::
+
+        ``flake8`` and ``isort`` are not mandatory, but ``black`` is.
+
+    .. note::
+
+        For ``flake8`` to play nicely with ``black``, you need to use a couple extra flags:
+
+        .. code-block:: bash
+
+            flake8 --max-line-length=88 --ignore=E203 path/to/file.py
+
+        ``--max-line-length=88`` is to match ``black``'s default line length.
+
+        ``--ignore=E203`` is to avoid conflicts between ``black`` and ``flake8``. ``black`` will add a space before ``:`` in slices, ``flake8`` will complain about it. This flag tells ``flake8`` to ignore this particular error.
```

Comment:
Do we want to have this note both here and later in the flake8 section?

Reply:
I thought about it but the point of the tl;dr is they get the minimum viable knowledge and I felt this was part of it. We can remove it if you want but I think it's fine

Reply:
No, it's fine with me, no strong opinion, I just wanted to flag it.